packages/client/src/Call.ts

```diff
@@ -766,38 +750,44 @@ export class Call {
     });
 
     // handlers for connection online/offline events
-    // Note: window.addEventListener is not available in React Native, hence the check
-    if (typeof window !== 'undefined' && window.addEventListener) {
-      const handleOnOffline = () => {
-        window.removeEventListener('offline', handleOnOffline);
-        this.logger('warn', '[Rejoin]: Going offline...');
-        this.state.setCallingState(CallingState.OFFLINE);
-      };
-
-      const handleOnOnline = () => {
-        window.removeEventListener('online', handleOnOnline);
-        if (this.state.callingState === CallingState.OFFLINE) {
-          this.logger('info', '[Rejoin]: Going online...');
-          rejoin().catch((err) => {
-            this.logger(
-              'error',
-              `[Rejoin]: Rejoin failed for ${this.reconnectAttempts} times. Giving up.`,
-              err,
-            );
-            this.state.setCallingState(CallingState.RECONNECTING_FAILED);
-          });
+    const unsubscribeOnlineEvent = this.streamClient.on(
+      'connection.changed',
+      (e) => {
+        if (e.type !== 'connection.changed') return;
+        if (!e.online) return;
+        unsubscribeOnlineEvent();
+        if (
+          [CallingState.JOINED, CallingState.JOINING].includes(
+            this.state.callingState,
+          )
+        ) {
+          return;
         }
-      };
-
-      window.addEventListener('offline', handleOnOffline);
-      window.addEventListener('online', handleOnOnline);
+        this.logger('info', '[Rejoin]: Going online...');
+        rejoin().catch((err) => {
+          this.logger(
+            'error',
+            `[Rejoin]: Rejoin failed for ${this.reconnectAttempts} times. Giving up.`,
+            err,
+          );
+          this.state.setCallingState(CallingState.RECONNECTING_FAILED);
+        });
+      },
+    );
+    const unsubscribeOfflineEvent = this.streamClient.on(
+      'connection.changed',
+      (e) => {
+        if (e.type !== 'connection.changed') return;
+        if (e.online) return;
+        unsubscribeOfflineEvent();
+        this.state.setCallingState(CallingState.OFFLINE);
+      },
+    );
```

Comment:
Perhaps you can merge the two event handlers in one

Reply:
We unfortunately cant.. the event listeners have to be unsubscribed as soon as it is fired once so that it doesn't live until the entire life of the call until call is left.

Reply:
oh, ok. Understood.